### PR TITLE
bump to spring-boot 2.5.1

### LINKED ISSUE #369: Upgrade to Spring Boot 2.5.1

### PR DESCRIPTION
**Pull request by pull-vert**
close #369

### CODE REVIEW

File: kofu/src/main/kotlin/org/springframework/fu/kofu/ConfigurationDsl.kt [hunk 1]
```diff
@@ -58,7 +59,7 @@ open class ConfigurationDsl(private val dsl: ConfigurationDsl.() -> Unit): Abstr
 	 */
 	inline fun <reified T : Any> configurationProperties(properties: T? = null, prefix: String = ""): T {
 		val bindedProperties = properties ?: FunctionalConfigurationPropertiesBinder(context).bind(prefix, Bindable.of(T::class.java)).orElseGet { BeanUtils.instantiateClass(T::class.java) }
-		context.registerBean<T>("${T::class.java.simpleName.toLowerCase()}ConfigurationProperties") {
+		context.registerBean<T>("${T::class.java.simpleName.lowercase(Locale.getDefault())}ConfigurationProperties") {
```
Comment (vladimirfx) [inline kofu/src/main/kotlin/org/springframework/fu/kofu/ConfigurationDsl.kt:62]: For what default locale is needed here?
Shouldn't it be simple lowercase()?

File: samples/jafu-reactive-data-r2dbc/build.gradle.kts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 plugins {
 	id("io.spring.dependency-management") version "1.0.11.RELEASE"
-	id("org.springframework.boot") version "2.4.4"
+	id("org.springframework.boot") version "2.5.0"
```
Comment (vladimirfx) [inline samples/jafu-reactive-data-r2dbc/build.gradle.kts:3]: 2.5.1 already released

File: kofu/src/main/kotlin/org/springframework/fu/kofu/ConfigurationDsl.kt (hunk 1 above)
Comment (pull-vert) [reply kofu/src/main/kotlin/org/springframework/fu/kofu/ConfigurationDsl.kt:62]: toLowerCase is now deprecated in Kotlin 1.5.0, its Dokka doc describe it as `Returns a copy of this string converted to lower case using the rules of the default locale.`, the deprecation replacement tip suggest to change to :
```
ReplaceWith("lowercase(Locale.getDefault())", "java.util.Locale")
```
That's why I didi it :)

lowercase() doc is a little different : `Returns a copy of this string converted to lower case using Unicode mapping rules of the invariant locale.`

File: samples/jafu-reactive-data-r2dbc/build.gradle.kts (hunk 2 above)
Comment (pull-vert) [reply samples/jafu-reactive-data-r2dbc/build.gradle.kts:3]: 12 days ago when I submitted this pull request 2.5.1 was not released ;) I am testing it with 2.5.1 and will update my PR very soon

File: kofu/src/main/kotlin/org/springframework/fu/kofu/ConfigurationDsl.kt (hunk 1 above)
Comment (vladimirfx) [reply kofu/src/main/kotlin/org/springframework/fu/kofu/ConfigurationDsl.kt:62]: I think lowercase() is exactly what needed in this case - locale insensitive transformation.

Comment (pull-vert) [reply kofu/src/main/kotlin/org/springframework/fu/kofu/ConfigurationDsl.kt]: I made the change

**Comment by sdeleuze** [comment]
Thanks!

**Pull request merged**